src/components/mask-input/mask-input.spec.ts

```diff
@@ -0,0 +1,498 @@
+import { html } from 'lit';
+import { elementUpdated, expect, fixture } from '@open-wc/testing';
+import sinon from 'sinon';
+import { defineComponents } from '../../index.js';
+import { MaskParser } from './mask-parser.js';
+import IgcMaskInputComponent from './mask-input';
+
+describe('Masked input', () => {
+  before(() => defineComponents(IgcMaskInputComponent));
+
+  const parser = new MaskParser();
+  const defaultPrompt = '_';
+  const defaultMask = 'CCCCCCCCCC';
+
+  const syncParser = () => {
+    parser.mask = masked.mask;
+    parser.prompt = masked.prompt;
+  };
+  const input = () =>
+    masked.shadowRoot!.querySelector('input') as HTMLInputElement;
+
+  let masked: IgcMaskInputComponent;
+
+  describe('', async () => {
+    beforeEach(async () => {
+      masked = await fixture<IgcMaskInputComponent>(
+        html`<igc-mask-input></igc-mask-input>`
+      );
+    });
+
+    it('sensible default values', async () => {
+      expect(masked.prompt).to.equal(defaultPrompt);
+      expect(masked.mask).to.equal(defaultMask);
+      expect(masked.value).to.be.undefined;
+      expect(input().placeholder).to.equal(parser.apply());
+    });
+
+    it('prompt character change (no value)', async () => {
+      masked.prompt = '*';
+      syncParser();
+
+      await elementUpdated(masked);
+      expect(input().placeholder).to.equal(parser.apply());
+    });
+
+    it('prompt character change (value)', async () => {
+      masked.value = '777';
+      await elementUpdated(masked);
+
+      masked.prompt = '*';
+      syncParser();
+      await elementUpdated(masked);
+
+      expect(input().value).to.equal(parser.apply(masked.value));
+    });
+
+    it('mask change (no value)', async () => {
+      masked.mask = 'CCCC';
+      syncParser();
+
+      await elementUpdated(masked);
+      expect(input().placeholder).to.equal(parser.apply());
+    });
+
+    it('mask change (value)', async () => {
+      masked.value = '1111';
+      await elementUpdated(masked);
+
+      masked.mask = 'CC CC';
+      syncParser();
+
+      await elementUpdated(masked);
+      expect(input().value).to.equal(parser.apply(masked.value));
+    });
+
+    it('placeholder is updated correctly', async () => {
+      const placeholder = 'Enter payment info';
+      syncParser();
+
+      masked.placeholder = placeholder;
+      await elementUpdated(masked);
+
+      expect(input().placeholder).to.equal(placeholder);
+
+      masked.placeholder = '';
+      await elementUpdated(masked);
+
+      expect(input().placeholder).to.equal(parser.apply());
+    });
+
+    it('empty value without literals', async () => {
+      expect(masked.value).to.be.undefined;
+    });
+
+    it('empty value with literals', async () => {
+      masked.withLiterals = true;
+      syncParser();
+      await elementUpdated(masked);
+
+      expect(masked.value).to.equal(parser.apply());
```

Comment:
For the validation bit, mostly thinking about the case where the input is empty, required and still with literals and if that can mess the required check. From what I can see that won't happen since it only affects the value getter. Still might simply look odd getting a mask back on an empty input - it's still _value_ with literals, but there's no value.

As for the value itself, it currently behaves like "with literals and prompts". Technically, we can expose more options, but I'm torn on this - can't think of too many scenarios where the literals are wanted without the full mask filled (which should be achievable through validation), and even on partially filled at least you get the same fixed length value which should be simple enough to handle. Honestly, taking opinions on this one, could do either way.